src/player.js

```diff
@@ -276,6 +278,28 @@ class Player extends FakeEventTarget {
   }
 
   /**
+   * @function _onPlay
+   * @return {void}
+   * @private
+   */
+  _onPlay(): void {
+    this._eventManager.unlisten(this, Html5Events.PLAY);
+    this.dispatchEvent(new FakeEvent(CustomEvents.FIRST_PLAY));
+  }
+
+  /**
+   * @function _onEnded
+   * @return {void}
+   * @private
+   */
+  _onEnded(): void {
+    this._eventManager.listen(this, Html5Events.PLAY, () => {
+      this._eventManager.unlisten(this, Html5Events.PLAY);
```

Comment:
I think this assumption is problematic.
This means on each ended we will register the PLAY event, so even if you do change media I think this will get triggered, no?